# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/src/AsyncPanZoomController.cpp
# Commit: 14eab025027a
# Full Hash: 14eab025027a09c77de45e0d0c910b2a37eec082
# Author: Timothy Nikkel <tnikkel@gmail.com>
# Date: 2021-03-12 15:32:35
# Regressor Bug: 1696802
# File Overlap Count: 1
# Description:
#   Bug 1696802. Add code to support double tap to zoom on mac. r=botond,mstange
#   
#   We already have a pref for double tap to zoom with is already enabled by default so it's kind of awkward to add another pref.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D107389
# ==============================================================================

diff -r 759586ace266 -r 14eab025027a gfx/layers/apz/src/AsyncPanZoomController.cpp
--- a/gfx/layers/apz/src/AsyncPanZoomController.cpp	Fri Mar 12 01:46:25 2021 +0000
+++ b/gfx/layers/apz/src/AsyncPanZoomController.cpp	Fri Mar 12 02:10:36 2021 +0000
@@ -2895,14 +2895,15 @@
   APZC_LOG("%p got a double-tap in state %d\n", this, mState);
   RefPtr<GeckoContentController> controller = GetGeckoContentController();
   if (controller) {
-    MOZ_ASSERT(GetCurrentTouchBlock());
+    MOZ_ASSERT(!GetCurrentInputBlock() || GetCurrentTouchBlock());
     if (mZoomConstraints.mAllowDoubleTapZoom &&
-        GetCurrentTouchBlock()->TouchActionAllowsDoubleTapZoom()) {
+        (!GetCurrentInputBlock() ||
+         GetCurrentTouchBlock()->TouchActionAllowsDoubleTapZoom())) {
       if (Maybe<LayoutDevicePoint> geckoScreenPoint =
               ConvertToGecko(aEvent.mPoint)) {
-        controller->HandleTap(TapType::eDoubleTap, *geckoScreenPoint,
-                              aEvent.modifiers, GetGuid(),
-                              GetCurrentTouchBlock()->GetBlockId());
+        controller->HandleTap(
+            TapType::eDoubleTap, *geckoScreenPoint, aEvent.modifiers, GetGuid(),
+            GetCurrentTouchBlock() ? GetCurrentTouchBlock()->GetBlockId() : 0);
       }
     }
     return nsEventStatus_eConsumeNoDefault;